### PR TITLE
[fix] Retool internal versions are not semver compliant, so ensure that our helm chart is able to support all version formats

### PR DESCRIPTION
**Pull request by jjlgao**

### CODE REVIEW

File: charts/retool/templates/_helpers.tpl
Comment (june-xu) [inline charts/retool/templates/_helpers.tpl:203]: what if the retool version is `pr-3.122.0-9492dbc` for an internal instance?

File: charts/retool/templates/deployment_backend.yaml
Comment (june-xu) [inline charts/retool/templates/deployment_backend.yaml:85]: reading this is tough in line, is there a helper file we can add this to lol

Comment (june-xu) [reply charts/retool/templates/deployment_backend.yaml:85]: an example of a helm function: https://github.com/tryretool/retool-k8s/blob/e053bbf66ad991dbed245adf745aa4605cded79d/terragrunt/src/managed-self-hosted/helm/retool-fork/charts/postgresql/charts/common/templates/_ingress.tpl#L36-L42 we can put it in a file called `_helpers.tpl`

**Review by mertbozfakioglu — COMMENTED** [review]
This LGTM but I'd need to defer to someone from infra for the approval

File: .github/workflows/ci.yaml
Comment (jjlgao) [inline .github/workflows/ci.yaml:30]: this is because github changed the ubuntu OS so now only python 3.9-3.12 are supported

**Review by june-xu — APPROVED** [review]
nice! thanks for the refactor

**Pull request merged**
